1.19.0-alpha.2

matteobruni opened the pull request:
feat: fixed window resize, particles are now moved to the right position. closes #890
feat: new out mode split added, it calls the destroy after bounce. fixes #994
refactor: rewritten emitters loops to work with the animation loop and timestamp delta
refactor: renamed backgroundMode to fullScreen, still supported the deprecated property
fix: fixed RecursivePartial type
refactor: infecter is a container singleton instead of a particle object

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.5.0 to 5.5.1

Bumps [webpack](https://github.com/webpack/webpack) from 5.5.0 to 5.5.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.5.0...v5.5.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump svelte-check from 1.1.12 to 1.1.13

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.1.12 to 1.1.13.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1034 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.5.1

build(deps-dev): bump webpack from 5.5.0 to 5.5.1

matteobruni committed:
Merge pull request #1035 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.1.13

build(deps-dev): bump svelte-check from 1.1.12 to 1.1.13

matteobruni committed:
build: angular updates

dependabot[bot] committed:
build(deps-dev): bump @angular/cli from 11.0.1 to 11.0.2

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 11.0.1 to 11.0.2.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v11.0.1...v11.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
refactor: little refactor on EmitterInstance.ts

matteobruni committed:
Merge pull request #1033 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-11.0.2

build(deps-dev): bump @angular/cli from 11.0.1 to 11.0.2

dependabot[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 0.1100.1 to 0.1100.2.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/commits)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1032 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-0.1100.2

build(deps-dev): bump @angular-devkit/build-angular from 0.1100.1 to 0.1100.2

matteobruni committed:
feat: fixed window resize, particles are now moved to the right position. closes #890
build: updated core main project to TS 4.1.2

ghost commented:
## DeepCode's analysis on [#c7225c](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/79529bd4788c7384e98109c4be34409841447058/matteobruni/tsparticles/c7225c2412ed68d66c5b99574253e1aef934ecb4/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=1&) found:

- :warning: **1** warning, :information_source: **1** minor issue. :point_down:
- :heavy_check_mark: **2** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">The static function clear should not be accessed with this. Use the class instance name instead. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/c7225c2412ed68d66c5b99574253e1aef934ecb4/core/main/src/Utils/CanvasUtils.ts#L40">CanvasUtils.ts:40</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/79529bd4788c7384e98109c4be34409841447058/matteobruni/tsparticles/c7225c2412ed68d66c5b99574253e1aef934ecb4/pr/_/%2Fcore%2Fmain%2Fsrc%2FUtils%2FCanvasUtils.ts/javascript%2Fdc_interfile_project%2FStaticAccessThis/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/c7225c2412ed68d66c5b99574253e1aef934ecb4/core/main/src/Options/Classes/Options.ts#L25">Options.ts:25</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/c7225c2412ed68d66c5b99574253e1aef934ecb4/core/main/src/Options/Classes/Options.ts#L34">Options.ts:34</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/c7225c2412ed68d66c5b99574253e1aef934ecb4/core/main/src/Options/Classes/Options.ts#L41">Options.ts:41</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/79529bd4788c7384e98109c4be34409841447058/matteobruni/tsparticles/c7225c2412ed68d66c5b99574253e1aef934ecb4/pr/_/%2Fcore%2Fmain%2Fsrc%2FOptions%2FClasses%2FOptions.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/79529bd4788c7384e98109c4be34409841447058/matteobruni/tsparticles/c7225c2412ed68d66c5b99574253e1aef934ecb4/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

dependabot[bot] committed:
build(deps-dev): bump jsoneditor from 9.1.2 to 9.1.3

Bumps [jsoneditor](https://github.com/josdejong/jsoneditor) from 9.1.2 to 9.1.3.
- [Release notes](https://github.com/josdejong/jsoneditor/releases)
- [Changelog](https://github.com/josdejong/jsoneditor/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/jsoneditor/compare/v9.1.2...v9.1.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.1.7 to 8.1.8

Bumps [postcss](https://github.com/postcss/postcss) from 8.1.7 to 8.1.8.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.1.7...8.1.8)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1037 from matteobruni/dependabot/npm_and_yarn/dev/jsoneditor-9.1.3

build(deps-dev): bump jsoneditor from 9.1.2 to 9.1.3

matteobruni committed:
build: fixed build

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
Merge pull request #1039 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.1.8

build(deps-dev): bump postcss from 8.1.7 to 8.1.8

matteobruni committed:
build: updated deps

matteobruni committed:
build: updated deps

matteobruni committed:
feat: new out mode split added, it calls the destroy after bounce. fixes #994

matteobruni committed:
build: fixed DestroyMode value names

matteobruni committed:
chore: improved fountain preset
fix: fixed destroy count on split

matteobruni committed:
build: updated deps on presets and shapes

matteobruni committed:
build: updated deps on components

matteobruni committed:
build: updated deps on components

matteobruni committed:
chore: improved fountain preset

matteobruni committed:
refactor: rewritten emitters loops to work with the animation loop and timestamp delta

matteobruni committed:
refactor: updaters now have only container in constructor parameters

matteobruni committed:
refactor: little refactoring for some core files

matteobruni committed:
refactor: renamed backgroundMode to fullScreen, still supported the deprecated property

matteobruni committed:
build: updated noconfig config

matteobruni committed:
docs: updating some docs files

dependabot[bot] committed:
build(deps-dev): bump eslint from 7.13.0 to 7.14.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.13.0 to 7.14.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.13.0...v7.14.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump vue-class-component from 7.2.6 to 8.0.0-rc.1

Bumps [vue-class-component](https://github.com/vuejs/vue-class-component) from 7.2.6 to 8.0.0-rc.1.
- [Release notes](https://github.com/vuejs/vue-class-component/releases)
- [Changelog](https://github.com/vuejs/vue-class-component/blob/v8.0.0-rc.1/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vue-class-component/compare/v7.2.6...v8.0.0-rc.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.1.8 to 8.1.9

Bumps [postcss](https://github.com/postcss/postcss) from 8.1.8 to 8.1.9.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.1.8...8.1.9)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/react from 16.14.0 to 17.0.0

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 16.14.0 to 17.0.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @babel/preset-env from 7.12.1 to 7.12.7

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.12.1 to 7.12.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.7/packages/babel-preset-env)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1046 from matteobruni/dependabot/npm_and_yarn/dev/types/react-17.0.0

build(deps-dev): bump @types/react from 16.14.0 to 17.0.0

matteobruni committed:
Merge pull request #1043 from matteobruni/dependabot/npm_and_yarn/dev/eslint-7.14.0

build(deps-dev): bump eslint from 7.13.0 to 7.14.0

matteobruni committed:
Merge pull request #1047 from matteobruni/dependabot/npm_and_yarn/dev/babel/preset-env-7.12.7

build(deps-dev): bump @babel/preset-env from 7.12.1 to 7.12.7

matteobruni committed:
Merge pull request #1045 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.1.9

build(deps-dev): bump postcss from 8.1.8 to 8.1.9

matteobruni committed:
Merge pull request #1044 from matteobruni/dependabot/npm_and_yarn/dev/vue-class-component-8.0.0-rc.1

build(deps): bump vue-class-component from 7.2.6 to 8.0.0-rc.1

matteobruni committed:
docs: updating more docs files

matteobruni committed:
build: fixed some types

matteobruni committed:
build: little code fixes

matteobruni committed:
fix: fixed RecursivePartial type

matteobruni committed:
refactor: infecter is a container singleton instead of a particle object

matteobruni committed:
fix: fixed background mask without a fill color

matteobruni committed:
chore(release): published new version

 - ng-particles@2.2.0-alpha.2
 - inferno-particles@1.2.0-alpha.2
 - jquery-particles@1.19.0-alpha.2
 - preact-particles@1.19.0-alpha.2
 - react-tsparticles@1.19.0-alpha.2
 - svelte-particles@1.2.0-alpha.2
 - particles.vue@2.2.0-alpha.2
 - particles.vue3@1.2.0-alpha.2
 - tsparticles-editor@1.19.0-alpha.2
 - tsparticles@1.19.0-alpha.2
 - ng-particles-demo@1.4.0-alpha.2
 - editor-demo@1.4.0-alpha.2
 - inferno-particles-demo@1.2.0-alpha.2
 - jquery-particles-demo@1.4.0-alpha.2
 - tsparticles-demo@1.4.0-alpha.2
 - preact-particles-demo@1.4.0-alpha.2
 - react-tsparticles-demo@1.4.0-alpha.2
 - svelte-demo@1.2.0-alpha.2
 - particles.vue-demo@1.4.0-alpha.2
 - particles.vue3-demo@1.2.0-alpha.2
 - tsparticles-preset-big-circles@1.4.0-alpha.2
 - tsparticles-preset-fire@1.4.0-alpha.2
 - tsparticles-preset-fountain@1.4.0-alpha.2
 - tsparticles-preset-links@1.4.0-alpha.2
 - tsparticles-preset-sea-anemone@1.4.0-alpha.2
 - tsparticles-preset-snow@1.4.0-alpha.2
 - tsparticles-preset-stars@1.4.0-alpha.2
 - tsparticles-shape-bubble@1.4.0-alpha.2
 - tsparticles-shape-heart@1.4.0-alpha.2
 - tsparticles-shape-multiline-text@1.4.0-alpha.2
 - tsparticles-shape-spiral@1.4.0-alpha.2
 - cra-template-particles-typescript@1.2.0-alpha.2
 - cra-template-particles@1.2.0-alpha.2
